include/odp/api/spec/ipsec.h

```diff
@@ -983,9 +983,55 @@ typedef struct odp_ipsec_op_flag_t {
  * These may be used to override some SA level options
  */
 typedef struct odp_ipsec_out_opt_t {
+	/** Union of all flag bits */
+	union {
+		/** Option flags. Set flag for those options that are
+		 *  used, all other options are ignored. */
+		struct {
+			/** Use fragmentation mode option */
+			uint32_t frag_mode: 1;
+
+			/** Use IP parameters option */
+			uint32_t ip_param:  1;
+
+			/** Use TFC padding length option */
+			uint32_t tfc_pad:   1;
+
+			/** Tunnel mode TFC dummy packet. In tunnel mode, set
+			 *  this flag to create a TFC dummy packet. The flag
+			 *  indicates packet data (at L3 offset) does not
+			 *  contain an inner packet IP header. If SA is
+			 *  configured to copy IP header fields from inner
+			 *  packet, those fields must be passed with
+			 *  IP parameters option. */
+			uint32_t tfc_dummy: 1;
+		} flag;
+
+		/** All flag bits */
+		uint32_t all_flags;
+	};
+
 	/** Fragmentation mode */
 	odp_ipsec_frag_mode_t frag_mode;
 
+	/** Union of IP parameters */
+	union {
+		/** Override IPv4 parameters in outer header creation.
+		 *  IP addresses are ignored. */
+		odp_ipsec_ipv4_param_t ipv4;
+
+		/** Override IPv6 parameters in outer header creation.
+		 *  IP addresses are ignored. */
+		odp_ipsec_ipv6_param_t ipv6;
+	};
+
+	/** TFC padding length
+	 *
+	 *  Number of TFC padding bytes added to the packet during IPSEC
+	 *  processing. Implementation guarantees that the padding does not
+	 *  contain any confidential information. */
```

Comment:
The implementation simply needs to ensure that it doesn't reuse application buffers, as they may contain residual confidential information. This can be done by inserting zeros or random bytes as the implementation chooses. Given the importance of security, it's appropriate to put this caveat in the spec as a reminder to implementations.

Reply:
This should be clear for implementers already from RFC, but it's also here since it's easy to forget that data cannot be sent out from any random address (e.g. end of an buffer). Implementation must be sure about the memory content (zeros, rand, counter) before including that as padding.